compiler/rustc_infer/src/infer/error_reporting/nice_region_error/util.rs

```diff
@@ -51,9 +52,19 @@ pub fn find_param_with_region<'tcx>(
     let hir_id = hir.local_def_id_to_hir_id(id.as_local()?);
     let body_id = hir.maybe_body_owned_by(hir_id)?;
     let body = hir.body(body_id);
+
+    // Don't perform this on closures
```

Comment:
*why*?

Reply:
Because we can't call  `fn_sig` on closures.

Reply:
In the resulting ICE if you call `fn_sig` on closures, it tells you what you should call instead: `substs.as_closure().sig()`

Reply:
I don't think it actually makes sense to look for closures here given the function's intention. This was previously never called on closures, it's just that the way we can call this we can still end up with a closure in this call and we need the HIR check to properly reject this if I remember correctly.

Reply:
I suspect that this might have been what caused the output changes that used to point at `async` and closures (but don't know for sure and the output changes aren't too dramatic).